### PR TITLE
Extra mutation lines

### PR DESCRIPTION
**Pull request by hyanwong**
## Description

Use pytest command-line option to create test SVG files, and add a (normally hidden) hidden line between a mutation and its node to allow nicer display of descendants of a mutation (garish test example given in `data/svg/ts_mut_highlight.svg`)

### CODE REVIEW

**Review by benjeffery — APPROVED** [review]
LGTM, couple small comments.

**Comment by jeromekelleher** [comment]
Looks good to me also, but seconding @benjeffery's comments.

Can you paste in an example of what the new viz looks like please?

File: python/tests/test_drawing.py [hunk 1]
```diff
@@ -1884,69 +1891,88 @@ def test_tree_root_branch(self):
...
             svg
         ).toprettyxml()  # Prettify for easy viewing
         self.verify_basic_svg(svg, **kwargs)
         svg_fn = pathlib.Path(__file__).parent / "data" / "svg" / filename
-        # TODO: replace below with a pytest flag
-        # with open(svg_fn, "wt") as file:  # Uncomment to save new expected files
-        #     file.write(svg)
+        if save:
+            logging.warning(f"Overwriting SVG file `{svg_fn}` with new version")
+            with open(svg_fn, "wt") as file:  # Uncomment to save new expected files
```
[the remark this answers is not in the document]
Comment (hyanwong) [reply python/tests/test_drawing.py:1903]: Oops. Thanks for spotting

**Comment by hyanwong** [comment]
> Looks good to me also, but seconding @benjeffery's comments.
> 
> Can you paste in an example of what the new viz looks like please?

There should be no change to the default, but if you want to highlight the descendants of a mutation, you can "unhide" a line descending from that mutation. The file at tests/data/svg/ts_mut_highlight.svg look like this (where I've changed some colours to make unhighlighted mutations pink, so they aren't so prominent, and you can see the colours of the focal mutations (red, blue, cyan) more easily. This particular example looks a bit garish, but you get the idea.


Screenshot 2021-03-02 at 09 12 25

File: python/tests/test_drawing.py [hunk 2]
```diff
@@ -1884,69 +1891,88 @@ def test_tree_root_branch(self):
         assert not ("H 0" in snippet2a)  # No root branch
         assert "H 0" in snippet2b
 
-    def verify_known_svg(self, svg, filename, **kwargs):
+    def verify_known_svg(self, svg, filename, save=False, **kwargs):
```
[the remark this answers is not in the document]
Comment (hyanwong) [reply python/tests/test_drawing.py:1894]: That SO page isn't clear to me @benjeffery . How do I get the value of `request.config.getoption("--overwrite-expected-visualizations")` within the `verify_known_svg(self, ...)` function? I've just tried a few things without success.

[the remark this answers is not in the document]
Comment (hyanwong) [reply python/tests/test_drawing.py:1894]: (If I just call it from within the verify function, I get a warning: "Fixture "overwrite_viz" called directly. Fixtures are not meant to be called directly,")

**Review by jeromekelleher — APPROVED** [review]

File: python/tests/test_drawing.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (benjeffery) [reply python/tests/test_drawing.py:1894]: Doh, sorry this was removed in pytest v4: https://docs.pytest.org/en/latest/deprecations.html#calling-fixtures-directly

**Pull request merged**